packages/ui-components/src/components/AppShellProvider/AppShellProvider.component.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from "react"
+
+import { StyleProvider } from "../StyleProvider/StyleProvider.component"
+import { ShadowRoot } from "../ShadowRoot/ShadowRoot.component"
+import { PortalProvider } from "../PortalProvider/PortalProvider.component"
+
+/**
+ * This provider acts as a wrapper for Juno apps. It renders a StyleProvider and PortalProvider
+ */
+export const AppShellProvider = ({
+  shadowRoot = true,
+  shadowRootMode = "open",
+  stylesWrapper = "inline",
+  theme = null,
+  children,
+}: AppShellProviderProps) => {
+  const Wrapper = React.useCallback(
```

Comment:
We now have the opportunity to refactor it properly by extracting it into a regular component, outside of this one, without the need for useCallback.